geps/gep-3779/index.md

```diff
@@ -241,23 +241,328 @@ spec:
+
+```go
+
+// PolicyTargetReferenceWithLabelSelectors specifies a reference to a set of Kubernetes
+// objects by Group and Kind, with an optional label selector to narrow down the matching
+// objects.
+//
+// Currently, we only support label selectors when targeting Pods.
+// This restriction is intentional to limit the complexity and potential
+// ambiguity of supporting label selectors for arbitrary Kubernetes kinds.
+// Unless there is a very strong justification in the future, we plan to keep this
+// functionality limited to selecting Pods only.
+//
+// This is currently experimental in the Gateway API and should only be used
+// for policies implemented within Gateway API. It is currently not intended for general-purpose
+// use outside of Gateway API resources.
+// +kubebuilder:validation:XValidation:rule="!(has(self.selector)) || (self.kind == 'Pod' && (self.group == '' || self.group == 'core'))",message="Selector may only be set when targeting Pods."
+type PolicyTargetReferenceWithLabelSelectors struct {
+  // Group is the group of the target object.
+  Group Group `json:"group"`
+
+  // Kind is the kind of the target object.
+  Kind Kind `json:"kind"`
+
+  // Selector is the label selector of target objects of the specified kind.
+  Selector *metav1.LabelSelector `json:"selector"`
+}
+
+```
+
+##### **Enhanced Discoverability with `gwctl`**
+
+A key challenge with `LabelSelector` is the loss of discoverability. It’s easier to see which policies target a `Service` but difficult to determine which policies might affect a specific pod.
+
+To address this, **investment in tooling is required.** Specifically, the `gwctl` CLI tool should be enhanced to provide insights such as:
+
+```sh
+TODO: complete gwctl commands
+```
+
+Without dedicated tooling, the `LabelSelector` approach could significantly degrade the user experience and observability.
+
+### API Design
+
+```go
+
+type AuthorizationPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of AuthorizationPolicy.
+    Spec AuthorizationPolicySpec `json:"spec,omitempty"`
+
+    // Status defines the current state of AuthorizationPolicy.
+    Status PolicyStatus `json:"status,omitempty"`
+}
+
+// AuthorizationPolicyAction specifies the action to take.
+// +kubebuilder:validation:Enum=ALLOW;DENY
+type AuthorizationPolicyAction string
+
+const (
+    // ActionAllow allows requests that match the policy rules.
+    ActionAllow AuthorizationPolicyAction = "ALLOW"
+    // ActionDeny denies requests that match the policy rules.
+    ActionDeny AuthorizationPolicyAction = "DENY"
+)
+
+// AuthorizationPolicySpec defines the desired state of AuthorizationPolicy.
+type AuthorizationPolicySpec struct {
+    // TargetRef identifies the resource this policy is attached to.
+    // +kubebuilder:validation:Required
+    TargetRefs []gatewayv1.PolicyTargetReferenceWithLabelSelectors `json:"targetRefs"`
```

Comment:
why do we need a list here? I think we only accept one pod right?

Reply:
TargetRef Pod = name must be empty, selector must be set, length must be 1
Otherwise multiple refs as long as target isn't Pods, selector can't be set

Reply:
Please add a CEL validaion for the same.